Algorithm Workbench (Question 3 & Question 4)

Question 3
Write a for loop that displays the following set of numbers: 
0,10,20,30,40,50 . . . 1000

In [9]:
# Loop through a range of numbers from 0 to 1000 with a step of 10
for i in range(0, 1001, 10):
    # Print the current number in the sequence
    print(i)

0
10
20
30
40
50
60
70
80
90
100
110
120
130
140
150
160
170
180
190
200
210
220
230
240
250
260
270
280
290
300
310
320
330
340
350
360
370
380
390
400
410
420
430
440
450
460
470
480
490
500
510
520
530
540
550
560
570
580
590
600
610
620
630
640
650
660
670
680
690
700
710
720
730
740
750
760
770
780
790
800
810
820
830
840
850
860
870
880
890
900
910
920
930
940
950
960
970
980
990
1000


Above we have the starting value (0), the stopping value (1001)(it ends one before), and the step value (10). As you can see its spaced in increments of 10  

Question 4 
Write a loop that asks the user to enter the number. the loop should iterate 10 times and keep a running total of the number entered

In [6]:
# Initialize the total to 0
total = 0

# Loop 10 times to get 10 numbers from the user
for _ in range(10):
    # Prompt the user to enter a number and convert it to a float
    number = float(input("Enter a number: "))
    # Add the entered number to the running total
    total += number

# Print the running total of the numbers entered
print(f"The running total of the numbers entered is: {total}")

Enter a number:  5
Enter a number:  10
Enter a number:  12
Enter a number:  20
Enter a number:  13
Enter a number:  34
Enter a number:  16
Enter a number:  26
Enter a number:  22
Enter a number:  9


The running total of the numbers entered is: 167.0


Again you can see above the code entered works since the range is 10 and you the total to the number which is what you input. Lastly you print a string then the final total calculated

Programming Excercises Questions 1,4,13, and 14

Programming Excercises - Question 1
A bug collector collects bugs every day for five days. Write a program that keeps a running total of the number of bugs collected during the five days. The loop should ask for the number of bugs collected for each day, and when the loop is finished, the program should display the total number of bugs collected. 

In [3]:
# Initialize the total number of bugs collected to 0
total_bugs = 0

# Loop over a range of 5 days (from 1 to 5)
for day in range(1, 6):
    # Ask the user to input the number of bugs collected for the current day
    bugs_collected = int(input(f"Enter the number of bugs collected on day {day}: "))
    # Add the number of bugs collected today to the running total
    total_bugs += bugs_collected

# After the loop, print the total number of bugs collected over five days
print(f"The total number of bugs collected over five days is: {total_bugs}")

Enter the number of bugs collected on day 1:  18
Enter the number of bugs collected on day 2:  12
Enter the number of bugs collected on day 3:  6
Enter the number of bugs collected on day 4:  7
Enter the number of bugs collected on day 5:  14


The total number of bugs collected over five days is: 57


First, above we see that we have to initialize the number of bugs from 0, then create a range from 1 to 5, then a string to question and input our variable, next we add up the bugs collected to get the total bugs over 5 days then you print your final result into a string. 

Programming Excercises - Question 4
the distance a vehicle travels can be calculated as follows:
distance = speed x time 

For Example, if a train travels 40 miles per hour for three hours, the distance traveled is 120 miles. Write a program that asks the user for the speed of a vechile (in miles per hour) ans the number of hours it has traveled. It should then use a loop to display the disctance the vechile has traveled wfor each hour of that time period. Here is an example of the desired output: 

What is the speed of the vehicle in mph? 40 [Enter]
How many hours has it traveled? 3 [Enter]
Hour_____________________Distance Traveled
1                               40
2                               80
3                               120

In [6]:
# Get the speed of the vehicle in mph from the user
speed = float(input("What is the speed of the vehicle in mph? "))

# Get the number of hours the vehicle has traveled from the user
hours = int(input("How many hours has it traveled? "))

# Print the table header
print("Hour\tDistance Traveled")
print("-------------------------")

# Loop over each hour and calculate the distance traveled
for hour in range(1, hours + 1):
    distance = speed * hour
    print(f"{hour}\t{distance}")

What is the speed of the vehicle in mph?  35
How many hours has it traveled?  2


Hour	Distance Traveled
-------------------------
1	35.0
2	70.0


Pretty simple question we have above us, all you do is use the formula given, then input the values to substitute for the variables in said formula then you have the correct answer/oupout 

Programming Excercises - (Question 13)
Print the pattern shown below

In [5]:
def draw_inverted_triangle():
    # Outer loop to control the number of lines, starting from 7 and decreasing to 1
    for i in range(7, 0, -1):
        # Inner loop to print the asterisks on each line
        for j in range(i):
            # Print asterisk without moving to the next line
            print("*", end="")
        # Move to the next line after printing all asterisks for the current line
        print()

# Call the function to draw the inverted triangle pattern
draw_inverted_triangle()

*******
******
*****
****
***
**
*


All you have to do is tell them the numbers of rows then tell them the rang e in which they should be outputted, since its a nested loop you go from outer loop to inner loop 

Programming Excercises - (Question 14)
Print the pattern shown 

In [4]:
def draw_pattern():
    # Outer loop to control the number of lines
    for i in range(6):
        # Print the first hash character at the beginning of the line
        print("#", end="")
        # Inner loop to print the spaces after the first hash character
        for j in range(i):
            print(" ", end="")
        # Print the second hash character at the end of the line
        print("#")

# Call the function to draw the pattern
draw_pattern()

##
# #
#  #
#   #
#    #
#     #


The question above is pretty much the same as the one before it (Question 13) with jsut different symbols, either way, like before, you start from the outer loop and make your way in, we create a range and for every new line we go one space to the right for the right side hastag. 